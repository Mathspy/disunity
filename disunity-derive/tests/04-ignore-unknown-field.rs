use disunity_derive::Variant;

#[derive(Variant)]
enum Class {
    Unknown(u32),
    #[disunity(discriminant = 1)]
    GameObject {
        field: i32,
    },
    #[disunity(discriminant = 2)]
    Transform,
}

fn main() {
    let _ = ClassVariant::GameObject;
    let _ = ClassVariant::Transform;
    let _ = ClassVariant::Unknown;
}
