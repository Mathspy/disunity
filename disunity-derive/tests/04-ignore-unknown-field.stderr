error[E0599]: no variant or associated item named `Unknown` found for enum `ClassVariant` in the current scope
  --> tests/04-ignore-unknown-field.rs:17:27
   |
3  | #[derive(Variant)]
   |          ------- variant or associated item `Unknown` not found here
...
17 |     let _ = ClassVariant::Unknown;
   |                           ^^^^^^^ variant or associated item not found in `ClassVariant`
